src/rules/maxLineLengthRule.ts

```diff
@@ -49,15 +68,28 @@ export class Rule extends Lint.Rules.AbstractRule {
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, {
+            ignoreUrls: this.ruleArguments.indexOf(OPTION_IGNORE_URL) > -1,
+            limit: this.ruleArguments[0] as number,
+        });
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (line.contentLength > ctx.options.limit) {
+            if (ctx.options.ignoreUrls) {               //
+                if (hasUrl(line, ctx.sourceFile)) {     // To minimize regex searches
+                    continue;
+                }
+            }
+            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(ctx.options.limit));
         }
     }
 }
+
+function hasUrl(line: LineRange, sourceFile: ts.SourceFile): boolean {
+    const lineText = sourceFile.text.substr(line.pos, line.contentLength);
+    const forUrl = /[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/g;
```

Comment:
You don't want to maintain this thing. just use a package like https://www.npmjs.com/package/url-regex

Reply:
I ran into a problem when trying to add the `url-regex` package. I'll look over this again soon. Sorry for the delay.

Reply:
I guess you have problems because there are no types included in the package and also nothing available as `@types/url-regex`?
You can simply define types for the module in this repository. I also had to do this: https://github.com/palantir/tslint/blob/master/src/custom-types.d.ts